src/lib.rs

```diff
@@ -2439,6 +2456,38 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool, target: &Target) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+
+        let mut cc_flags: Vec<OsString> = env_os
+            .to_string_lossy()
+            .split("\u{1f}")
+            // Strip prefixes from rustc flags
+            .flat_map(|flag| {
+                if flag == "-Z" || flag == "-C" {
+                    None
+                } else if flag.starts_with("-Z") || flag.starts_with("-C") {
+                    Some(&flag[2..])
+                } else {
+                    Some(flag)
+                }
+            })
+            .flat_map(|flag| rustc_to_cc_flag(flag, cmd.family))
+            // Filter out flags not supported by the currently used CC
+            .filter(|flag| {
+                self.is_flag_supported_inner(flag, &cmd.path, &target)
+                    .unwrap_or(false)
+            })
```

Comment:
I think we may want to avoid this for several reasons:
1. Performance; I'm not sure how many flags Cargo sets by default (?), but each is an extra compiler invocation that could probably have been avoided.
2. Correctness; Setting `-Csoft-float=yes` changes the ABI, so it is (from my understanding?) _required_ for soundness that the corresponding `-msoft-float` flag is passed. If the C compiler doesn't support it, it should abort compilation, not generate something that won't work.

Reply:
> I'm not sure how many flags Cargo sets by default

`CARGO_ENCODED_RUSTFLAGS` is empty by default so those will all be user-provided flags, meaning not too many in practice. I think the benefit of knowing whether the used compiler supports the flag outweighs the slight performance overhead here?

> If the C compiler doesn't support it, it should abort compilation, not generate something that won't work.

I see the point, yeah. Maybe we could just generate a warning if we find out a flag isn't supported? Usually these kinds of flags come to clang and gcc before they come to Rust so this filtering here is mainly here so that the code isn't too picky with compiler versions. Otherwise I could refactor to check if the flag is supported within the big match statement so that we can decide whether to skip, warn or abort on a per-flag basis.